packages/services/src/contents/index.ts

```diff
@@ -254,6 +261,17 @@ export namespace Contents {
      * It should return `undefined` if the factory is not able to create a `ISharedDocument`.
      */
     createNew(options: ISharedFactoryOptions): ISharedDocument | undefined;
+
+    /**
+     * Register a SharedDocumentFactory.
+     *
+     * @param type Document type
+     * @param factory Document factory
+     */
+    registerDocumentFactory(
+      type: Contents.ContentType,
+      factory: SharedDocumentFactory
+    ): void;
```

Comment:
Adding a note that this is just exposing a function that was already implemented to simplify the code on the jupyterlab-collaboration side. Technically this is a breaking change, so maybe it should have an `?` added for now?